### PR TITLE
Start trying to improve testing options

### PR DESCRIPTION
**Pull request by hamishcoleman**
without installing a bucket-load of virtualenv stuff that is not tracked or managed by my selected package mangement tool.

Still has at least the following issue:
- the version of python3-django-ratelimit is too old in debian sid (which has been bypassed with some compile-time hackery)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+    python3-django-redis \
+    python3-ipy \
+    python3-django-ratelimit \
+    python3-django-cors-headers \
+    python3-django-oauth-toolkit \
+
+# TODO - from the requirements.txt
+# pyserial>=3.2.1
+# RPi.GPIO>=0.6.3
+# paho-mqtt>=1.2
```
Comment (adisbladis) [inline Makefile]: mqtt could be dropped completely, it was never put in use anyway

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
+# raven>=6.0.0
```
Comment (adisbladis) [inline Makefile]: raven is no longer used

File: Makefile (hunk 1 above)
Comment (hamishcoleman) [reply Makefile]: yeah, I thought I remembered that

File: Makefile (hunk 2 above)
Comment (hamishcoleman) [reply Makefile]: cool, removed

**Comment by hamishcoleman** [comment]
Hey @adisbladis, any concerns about merging this now?

File: .gitignore [hunk 3]
```diff
@@ -4,3 +4,6 @@ db.sqlite3
 .cache
 .coverage
 hackman/static/admin
+
+# compiled objects
+*.pyc
```
Comment (adisbladis) [inline .gitignore]: `.pyc` is a python2 thing. `hackman` only runs on python3.

On python3 all `.pyc` files are collected in `__pycache__` which is already ignored.

You can also export the environment variable `PYTHONDONTWRITEBYTECODE` to not write these files in the first place.

In any case this at worst does nothing so it's fine.

File: Makefile [hunk 4]
```diff
@@ -1,9 +1,50 @@
...
+    python3-ipy \
+    python3-django-ratelimit \
+    python3-django-cors-headers \
+    python3-django-oauth-toolkit \
+    redis-server \
+    gunicorn3 \
+    python3-gevent \
+
+# TODO - from the requirements.txt
+# pyserial>=3.2.1
```
Comment (adisbladis) [inline Makefile]: Fyi this one is called `python3-serial` in debian-speak.

File: Makefile [hunk 5]
```diff
@@ -1,9 +1,50 @@
...
+# RPi.GPIO>=0.6.3
```
Comment (adisbladis) [inline Makefile]: This one is called `rpi.gpio` on `raspbian`.
It's only needed if your runtime env is a raspberry pi.

For testing there is a mock implementation: https://github.com/dimsumlabs/hackman/blob/8019b23736518573efd0444dd2a6a8ac3c00c8f6/hackman_door/door/impls/dummy_test.py

File: .gitignore (hunk 3 above)
Comment (hamishcoleman) [reply .gitignore]: Ha!  I knew that hackman was python3 only, but (as you can see from the Makefile) my pytest was running the python2 version (until I yelled at it).

I didnt know until I read your message that python3 had changed how they cache the compiled objects..

**Pull request merged**